gtk: Show release date for fwupd updates

crawfxrd opened the pull request:
fwupd stores the release date as an unsigned 64-bit Unix timestamp, whereas system76-firmware stores it as a string. Convert the System76 value to a timestamp to match fwupd.

Ref: https://github.com/pop-os/firmware-manager/pull/127

leviport commented:
Is it expected for all of the dates to default to the start of the Unix epoch? 

![2023-06-15_12-43](https://github.com/pop-os/firmware-manager/assets/13512097/71cb6cca-3c22-4241-834a-a71854e307b0)

If it's because a date isn't added, then maybe the date should only be shown if that info is present.

crawfxrd commented:
I'd expect to only see Unix epoch on System76 provided firmware and only if my logic for converting from a string to a timestamp is wrong.

For LVFS provided firmware, this just reads the `created` field for each release (a required field). It showing the Unix epoch means something is wrong with fetching the LVFS data (fwupd-dbus?).

crawfxrd committed:
gtk: Show release date for fwupd updates

fwupd stores the release date as an unsigned 64-bit Unix timestamp,
whereas system76-firmware stores it as a string. Convert the System76
value to a timestamp to match fwupd.

Signed-off-by: Tim Crawford <tcrawford@system76.com>

jacobgkau commented:
Rebased for testing.

jacobgkau reviewed (approved):
The dates are now displaying as expected. Before:
<img width="1920" height="1080" alt="Screenshot from 2025-07-16 15-51-34" src="https://github.com/user-attachments/assets/bf2fd5f1-7a9c-4993-a32c-fc95e091024a" />
After:
<img width="1920" height="1080" alt="Screenshot from 2025-07-16 15-52-49" src="https://github.com/user-attachments/assets/1b8f0782-00cf-4ca3-873d-d88ad6dd7e20" />
This was with a Logitech receiver rather than a Launch, but both have firmware coming from LVFS.

jacobgkau commented:
Actually, @leviport it might be good to take a quick look on a Launch if System76 and non-System76 stuff is handled differently. My testing was with a Logitech receiver.

leviport reviewed (approved):
Launch is looking good as well
<img width="1410" height="869" alt="Screenshot from 2025-07-17 11-23-06" src="https://github.com/user-attachments/assets/959bb0c4-d900-400d-b46a-6cf682707633" />
